### DISCUSSION TITLE
Permanent Top | Bottom Object

### DISCUSSION THREAD

**Discussion by maoweifan** [fabricjs/fabric.js#7760] [Q&A]
I think I have met a requirement. In the `canvas`, there are two `Objects`, one is always bottomed as the `background`, the other is always topped as the`UI operation`, and then all its content is in the middle of the two. In `fabric`, this is not very friendly, (maybe I didn't find the right way). After an `Object` is set to `bringToFront`, `canvasContext.add(Object)`, the `Object` will still be added to the top of the set `bringToFront`, so at this time, you need to set `bringToFront` again. I think such an operation is cumbersome. Do you have any good ideas 💡

**Comment by maoweifan** [comment]
like `setBackgroundColor` `setBackgroundImage`  `setOverlayColor`  `setOverlayImage`

**Comment by maoweifan** [comment]
I have a preliminary idea:
```javascript
const ctx = new fabric.Canvas(...)
const backgroundObject = new fabric.Rect({width: 200, height: 200, fill: 'white', left: 0, top: 0})
ctx.setBackgroundObject(backgroundObject)
```

**Comment by ShaMan123** [comment]
I'm not sure I understand what you need but you can easily leverage `backgroundImage` for your needs.
```js
canvas.backgroundImage = bottomObject;
canvas.overlayImage = topObject
```

**Reply by ShaMan123** [reply]
@asturur  It is a good idea to rename these props
We have broken them so let's break them entirely and rename

**Comment by ShaMan123** [comment]
I am not sure why you use `bringToFront`...
Another thing is the `add` method which adds an object to the top of the stack.
You can use `insertAt` to insert an object wherever you want.
